crates/consensus/src/engine/mod.rs

```diff
@@ -194,9 +202,29 @@ impl<Client: HeaderProvider + BlockProvider> ConsensusEngine for EthConsensusEng
             })
         }
 
-        // TODO: execute block
-
-        Ok(PayloadStatus::new(PayloadStatusEnum::Valid, block.hash()))
+        let (header, body, _) = block.split();
+        let transactions = body
+            .into_iter()
+            .map(|tx| {
+                let tx_hash = tx.hash;
+                tx.into_ecrecovered().ok_or(EngineApiError::PayloadSignerRecovery { hash: tx_hash })
+            })
+            .collect::<Result<Vec<_>, EngineApiError>>()?;
+        let state_provider = SubState::new(State::new(&*self.client));
+        let config = self.config.clone().into();
```

Comment:
Do we need this clone? Given we pass the Config by ref below we should be able to work with From<&T>?